### ISSUE TITLE
AWS::CloudFront::ResponseHeadersPolicy: add handler-enforced constraints missing from schema (Comment maxLength 128, AccessControl*.Items minItems 1)

### ISSUE THREAD

**Issue by esetnik** [aws-cloudformation/cfn-lint#4578]
### Summary

The provider schema for `AWS::CloudFront::ResponseHeadersPolicy` documents two **create-time constraints only in prose**, not as JSON Schema, so cfn-lint can't flag them. A template that violates either passes `cfn-lint` (and `aws cloudformation validate-template`) but **fails `aws cloudformation deploy`** at the resource handler.

### The two constraints

1. **`ResponseHeadersPolicyConfig.Comment`** — the schema `description` says *"The comment cannot be longer than 128 characters"* but there is no `maxLength`. Deploy error:
   > `Invalid request provided: AWS::CloudFront::ResponseHeadersPolicy: The parameter Comment is too big.`

2. **`AccessControlAllowHeaders.Items`** (and the siblings `AccessControlAllowMethods` / `AccessControlAllowOrigins` / `AccessControlExposeHeaders`) — `required`, but there is no `minItems`. An empty `Items: []` is valid per the schema — and is exactly what the AWS *managed* `Managed-CORS-with-preflight-and-SecurityHeadersPolicy` carries — yet it is rejected when **creating** a custom policy. Deploy error:
   > `Invalid request provided: AWS::CloudFront::ResponseHeadersPolicy: The parameter Allow Headers needs to have at least one item.`

### Repro (both pass `cfn-lint 1.48.1`, both fail at deploy)

```yaml
Resources:
  P:
    Type: AWS::CloudFront::ResponseHeadersPolicy
    Properties:
      ResponseHeadersPolicyConfig:
        Name: repro
        # A Comment longer than 128 characters -> "The parameter Comment is too big."
        Comment: >-
          this comment is deliberately far longer than one hundred and twenty eight characters so that the cloudfront resource handler rejects it at create time
        CorsConfig:
          AccessControlAllowCredentials: false
          AccessControlAllowHeaders:
            Items: []            # -> "The parameter Allow Headers needs to have at least one item."
          AccessControlAllowMethods:
            Items: [GET, HEAD]
          AccessControlAllowOrigins:
            Items: ["*"]
          OriginOverride: false
```

`cfn-lint template.yaml` exits `0`; `aws cloudformation deploy` fails on each constraint in turn.

### Root cause & suggested fix

Both limits live only in the AWS resource-provider schema's `description` text (`describe-type --type-name AWS::CloudFront::ResponseHeadersPolicy` shows no `maxLength`/`minItems` anywhere), so it is ultimately an upstream schema gap. But cfn-lint already carries an extension patch for this resource that encodes a handler constraint the base schema omits:

`data/schemas/patches/extensions/all/aws_cloudfront_responseheaderspolicy/smithy.json`
```json
[{"op":"add","path":"/definitions/AccessControlAllowMethods/properties/Items/items/enum","value":["ALL","DELETE","GET","HEAD","OPTIONS","PATCH","POST","PUT"]}]
```

Suggest extending that same patch with:
- `maxLength: 128` on `/definitions/ResponseHeadersPolicyConfig/properties/Comment`
- `minItems: 1` on `Items` of `AccessControlAllowHeaders`, `AccessControlAllowMethods`, `AccessControlAllowOrigins`, and `AccessControlExposeHeaders`

### Environment

- cfn-lint 1.48.1
